examples/ants.jl

```diff
@@ -0,0 +1,457 @@
+"""
+function agent_forage!(agent::Ant, model)
+    agents = collect(agents_in_position(agent.pos, model))
+    dinner_array = filter!(x -> isa(x, Food), agents)
+
+    if !isempty(dinner_array)
+        food = dinner_array[0]
+        eat!(agent, food, model)
+        agent.drop = model.init_drop
+        agent.state = :going_home
+    else
+        foraging_move!(agent, model)
+    end
+end
+
+"""
+Homing action, the ant will move one step closer to the colony. 
+It also drops pheromones along the way
+
+If the ant is on the colony, it drops the carry and goes back to foraging
+"""
+function agent_go_home!(agent::Ant, model)
+    agents = collect(agents_in_position(agent.pos, model))
+    colony_array = filter!(x -> isa(x, Colony), agents)
+
+    if isempty(colony_array)
+        drop_pheromone!(agent, model)
+        homing_move!(agent, model)
+    else
+        colony = colony_array[1]
+        colony.amount += agent.amount
+
+        agent.amount = 0
+        agent.angle = rotate_angle(agent.angle, pi) # turn back
+        agent.state = :foraging
+    end
+end
+
+"""
+The ant drops a pheromone on the current position, and reduces the amount of pheromone it will drop next
+"""
+function drop_pheromone!(agent, model)
+    model.ant_pheromone[agent.pos...] += agent.drop
+    agent.drop *= model.drop_rate
+end
+
+"""
+the ant takes an amount of 1 of the food source, if then the food is depleted, the food agent is deleted
+"""
+function eat!(agent::Ant, food, model)    
+    agent.amount += 1
+    food.amount -= 1
+    
+    if food.amount<=0
+        kill_agent!(food, model)
+    end
+end
+
+"""
+Agent step for the other agent types, blank because they do no action
+"""
+function agent_step!(_, _) 
+end
+
+# model actions
+"""
+Step for the model, diffuses the pheromones
+"""
+function model_step!(model)
+    diffuse!(model)
+end
+
+"""
+Diffuses pheromones giving 1/8 of the `diffusion_r*ant_pheromone` in the tile to each neighbor
+
+Also evaporates the pheromones with rate `1 - evaporation_r`
+"""
+function diffuse!(model)
+    diffusion_kernel = (1/8) * model.diffusion_r .* ones(3,3)
+    diffusion_kernel[2,2] = 1 - model.diffusion_r
+
+    diffused_grid = imfilter(model.ant_pheromone, diffusion_kernel) # diffuse with kernel
+    diffused_grid = (1-model.evaporation_r) .* diffused_grid  # evaporate
+    diffused_grid[diffused_grid.<= 0.05] .= 0  # if its too low make it zero
+    diffused_grid[model[0].pos...] = 0  # no pheromones inside the colony
+    
+    @inbounds for p in positions(model) # we don't have to enable bound checking
+        model.ant_pheromone[p...] = diffused_grid[p...]
+    end
+end
+
+# Plotting
+
+groupcolor(_::Ant) = :black;
+groupcolor(_::Colony) = RGBA(24/255, 48/255, 60/255, 0.6);
+groupcolor(a::Food) = @match a.amount begin 
+        0 => :black
+        _ => :green
+    end;
+
+groupmarker(a) = @match a begin
+        a::Ant => '⚫'
+        c::Colony => '■'
+        f::Food => '■'
+    end;
```

Comment:
Please avoid this. Don't use `Match` and instead write standard Julia code that does the same thing. Not only it is more readable, it even takes less lines of code.
```julia
groupmarker(a::Ant) = '⚫'
groupmarker(c::Colony) = '■'
groupmarker(f::Food) = '■'
```

Reply:
oh ok, ill change that part

Reply:
changed